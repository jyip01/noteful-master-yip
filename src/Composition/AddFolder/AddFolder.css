.formSideBar{
    padding-top:20px;
}

.formTitle{
    font-size:22px;
}

.form-group label{
    margin-right:10px;
}

.form-group #folderName{
    border:1px solid grey;
    padding:3px;
}

.form-group{
    margin-top:15px;
    margin-bottom:15px;
}

.error{
    color:red;
    font-size:12px;
}

.formSubmit{
    width:30%;
    display:block;
    height:40px;
    font-size:16px;
}

.formSubmitLink{
    text-decoration: none;
}

.disabled{
    background-color:grey;
    color:black;
    border:1px solid grey;
}

.notDisabled{
    border:1px solid black;
}

.notDisabled:hover{
    background-color:black;
    color:white;
    cursor: pointer;
}
