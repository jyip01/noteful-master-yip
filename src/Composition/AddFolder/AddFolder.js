import React, {Component} from 'react'
import ValidationError from './ValidationError'
import ApiContext from '../../ApiContext'
import config from '../../config'
import {Link} from 'react-router-dom'
import BackButton from '../BackButton/BackButton'
import './AddFolder.css'
import Error from '../../Error.js'


class AddFolder extends Component{
    static contextType = ApiContext;

    validateFolderName(){
        const folderName = this.state.folderName.value
        if(folderName.length===0){
            return 'Folder name is required.' 
        }
    }

    constructor(props){
        super(props)
        this.state={
            folderName: {
                value: '',
                touched: false
            }
        }
    }

    updateFolderName(folderName){
        this.setState({folderName: {value:folderName,touched:true}})
    }

    makeSubmitButton(){
        if(this.validateFolderName()){
            return <Link onClick={()=>{this.context.setFolder({})}} className="formSubmitLink" to='/'><button  disabled={this.validateFolderName()} className="formSubmit disabled" type="submit" onClick={()=>this.addFolderRequest(this.state.folderName.value, this.context.addFolder)}>Submit</button></Link>
        }
        else{
            return <Link onClick={()=>{this.context.setFolder({})}} className="formSubmitLink" to='/'><button className="formSubmit notDisabled" type="submit" onClick={()=>this.addFolderRequest(this.state.folderName.value, this.context.addFolder)}>Submit</button></Link>
        }
    }

    addFolderRequest(folderName, callback){
        fetch(`${config.API_ENDPOINT}/folders`,{
            method: 'POST',
            headers: {
                'content-type':'application/json',
            },
            body: JSON.stringify({"name":folderName})
        })
        .then(res=>{
            if(!res.ok){
                return res.json().then(error=>{
                    throw error
                })
            }
            return res.json()
        })
        .then(data=>{
            callback(data)

        })
        .catch(error=>{
            console.error(error)
        })
    }


    render(){

        return(
            <ApiContext.Consumer>
                {(context)=>(<Error>
                <div className="contentContainer">
                    <div className="sideBar formSideBar">
                        <BackButton/>
                    </div>
                    <div className="main">
                        <form className="newFolder">
                            <p className="formTitle">Create a New Folder</p>
                            <div className="form-group">
                                <label htmlFor="folderName">Folder Name*</label>
                                <input type="text" name="folderName" id="folderName" onChange={e=>this.updateFolderName(e.target.value)}/>
                                {this.state.folderName.touched && (<ValidationError message={this.validateFolderName()}/>)}
                            </div>
                            {this.makeSubmitButton()}    
                        </form>
                    </div>
                    
                </div>
                </Error>
                )}
            </ApiContext.Consumer>
            
        )
    }
}

export default AddFolder

