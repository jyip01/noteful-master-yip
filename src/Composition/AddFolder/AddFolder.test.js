import React from 'react'
import ReactDOM from 'react-dom'
import AddFolder from './AddFolder'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'


it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><AddFolder/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AddFolder/>
        </BrowserRouter>
      )
      .toJSON();
     expect(tree).toMatchSnapshot();
     });
