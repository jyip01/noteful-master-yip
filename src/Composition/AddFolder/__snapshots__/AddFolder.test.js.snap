// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<div
  className="contentContainer"
>
  <div
    className="sideBar formSideBar"
  >
    <a
      className="backButton"
      href="/"
      onClick={[Function]}
    >
      Go back
    </a>
  </div>
  <div
    className="main"
  >
    <form
      className="newFolder"
    >
      <p
        className="formTitle"
      >
        Create a New Folder
      </p>
      <div
        className="form-group"
      >
        <label
          htmlFor="folderName"
        >
          Folder Name*
        </label>
        <input
          id="folderName"
          name="folderName"
          onChange={[Function]}
          type="text"
        />
      </div>
      <a
        className="formSubmitLink"
        href="/"
        onClick={[Function]}
      >
        <button
          className="formSubmit disabled"
          disabled="Folder name is required."
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </a>
    </form>
  </div>
</div>
`;
